### PR TITLE
feat: voting logic

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

Closes https://app.clickup.com/t/86dqpcjn7

Includes the model, migration, method for adding votes and controller endpoint for voting

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: database/migrations/2023_12_01_161507_create_collection_votes_table.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::create('collection_votes', function (Blueprint $table) {
+            $table->id();
+            $table->foreignIdFor(Wallet::class, 'wallet_id')->constrained();
+            $table->foreignIdFor(Collection::class, 'collection_id')->constrained();
+            $table->timestamp('voted_at')->nullable();
+            $table->timestamps();
```
Comment (shahin-hq) [inline database/migrations/2023_12_01_161507_create_collection_votes_table.php:20]: I'd not add timestamps as there will be no update and `voted_at` already mimics `created_at`, wdyt?

Comment (alfonsobries) [reply database/migrations/2023_12_01_161507_create_collection_votes_table.php:20]: @shahin-hq makes sense

**Comment by alfonsobries** [comment]
@shahin-hq added `has_won_at`

**Comment by ItsANameToo** [comment]
https://app.clickup.com/t/86dqpzq4p follow up for seeder states

**Pull request merged**